backend/worker/kafka_worker.go

```diff
@@ -284,11 +284,8 @@ func (k *KafkaBatchWorker) flushTraces(ctx context.Context) error {
 			case lastMsg = <-k.BatchBuffer.messageQueue:
 				switch lastMsg.Type {
 				case kafkaqueue.PushTraces:
-					traceRows = append(traceRows, lastMsg.PushTraces.TraceRows...)
-					received += len(lastMsg.PushTraces.TraceRows)
-				}
-				if received >= k.BatchFlushSize {
-					return
```

Comment:
i believe we still want to keep this. what this does is make sure that if another go routine keeps adding to the `messageQueue` as we pull off of it (line 284), we'll stop once we hit the flush size